src/directives/analytics.js

```diff
@@ -1,16 +1,13 @@
 const MailDirective = {
-  beforeMount(el, binding, vnode) {
+  beforeMount(el, binding) {
     el.myAnalyticsHandler = () => {
-      if (!vnode.context) {
-        return
-      }
-      const matomo = vnode.context.$matomo
+      const matomo = binding?.instance?.$matomo
```

Comment:
https://v3.vuejs.org/guide/migration/custom-directives.html#edge-case-accessing-the-component-instance